Dockerfile

```diff
@@ -1,72 +1,50 @@
 FROM python:3.8-buster as python-base
-
 LABEL maintainer="TACC-ACI-WMA <wma_prtl@tacc.utexas.edu>"
-
 ARG DEBIAN_FRONTEND=noninteractive
-
-ENV PYTHONUNBUFFERED 1
-
-# https://python-poetry.org/docs/configuration/#using-environment-variables
-ENV POETRY_VERSION=1.4.0 \
-    POETRY_HOME="/opt/poetry" \
-    POETRY_VIRTUALENVS_IN_PROJECT=true \
-    POETRY_NO_INTERACTION=1 \
-    PYSETUP_PATH="/opt/pysetup" \
-    VENV_PATH="/opt/pysetup/.venv"
-
-# prepend poetry and venv to path
-ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
-
-FROM python-base as builder-base
-
 RUN apt-get update && apt-get install -y \
     build-essential python3-dev \
     libldap2-dev libsasl2-dev ldap-utils tox \
     lcov valgrind vim \
     && pip3 install uwsgi
 
-RUN pip3 install --upgrade pip setuptools wheel
+ENV PYTHONUNBUFFERED 1
+ENV PATH="/root/.local/bin:$PATH"
+
+# https://python-poetry.org/docs/configuration/#using-environment-variables
+ENV POETRY_VERSION=1.4.0 \
+    POETRY_VIRTUALENVS_CREATE=false \
+    POETRY_NO_INTERACTION=1
 
+RUN pip3 install --upgrade pip setuptools wheel
 # Install Poetry - respects $POETRY_VERSION & $POETRY_HOME
 RUN curl -sSL https://install.python-poetry.org | python3 -
-
+RUN mkdir /code
 # copy project requirement files here to ensure they will be cached.
-WORKDIR $PYSETUP_PATH
-COPY pyproject.toml poetry.lock ./
-
+COPY pyproject.toml poetry.lock /code/
+WORKDIR /code
 # install runtime deps - uses $POETRY_VIRTUALENVS_IN_PROJECT internally
 RUN poetry install --no-dev
 
 # `development` image is used for local development
 FROM python-base as development
-WORKDIR $PYSETUP_PATH
-
-# copy in our built poetry + venv
-COPY --from=builder-base $POETRY_HOME $POETRY_HOME
-COPY --from=builder-base $PYSETUP_PATH $PYSETUP_PATH
-
 # quicker install as runtime deps are already installed
+COPY . /code/
 RUN poetry install
 
-# `production` image is used for deployed runtime environments
-FROM python-base as production
-
-# install node
-RUN curl -sL https://deb.nodesource.com/setup_17.x | bash -
-RUN apt-get install -y nodejs
+FROM node:18 as node_build
+COPY package.json package-lock.json /code/
+WORKDIR /code
+RUN npm ci
 
-COPY --from=builder-base $PYSETUP_PATH $PYSETUP_PATH
+COPY . /code/
```

Comment:
Ah, I think I get it. First copy node dep's list in and build, then copy the rest in. This way, changes to the rest of the code don't trigger layer change thus don't cause `npm ci` for unrelated code changes.